pkg/core/resources.go

```diff
@@ -88,20 +86,28 @@ const (
 	AbstractConstructProvider = "klotho"
 )
 
-func (cid ConstructId) ToRid() ResourceId {
-	return ResourceId{
-		Provider: AbstractConstructProvider,
-		Type:     cid.Capability,
-		Name:     cid.ID,
-	}
-}
-
-func IsConstructOfCapability(baseConstruct BaseConstruct, cap string) bool {
+func IsConstructOfAnnotationCapability(baseConstruct BaseConstruct, cap string) bool {
 	cons, ok := baseConstruct.(Construct)
 	if !ok {
 		return false
 	}
-	return cons.Provenance().Capability == cap
+	return cons.AnnotationCapability() == cap
+}
+
+func ListAllConstructs() []Construct {
```

Comment:
This seems brittle and error prone, but I don't have a better suggestion right now.